core/modules/node/node.module

```diff
@@ -947,19 +947,27 @@ function node_validate($node, $form, &$form_state) {
   }
 
   // Validate the "Authored on" field.
-  if (!empty($node->date) && is_string($node->date) && strtotime($node->date) === FALSE) {
-    form_set_error('date', t('You have to specify a valid date.'));
+  if (!empty($node->date)) {
+    if (is_array($node->date)) {
+      $node_date = implode(' ', $node->date);
+    }
+    if (strtotime($node_date) === FALSE) {
+      form_set_error('date', t('You have to specify a valid date.'));
+    }
   }
 
   // Validate the "Publish on" date.
-  if ($node->status == NODE_SCHEDULED && !empty($node->scheduled) && is_string($node->scheduled)) {
+  if ($node->status == NODE_SCHEDULED && !empty($node->scheduled)) {
+    if (is_array($node->scheduled)) {
+      $node_scheduled = implode(' ', $node->scheduled);
+    }
     // Validate the entire string is valid.
-    if (strtotime($node->scheduled) === FALSE) {
+    if (strtotime($node_scheduled) === FALSE) {
```

Comment:
`$node_scheduled` needs to exist outside the previous if as well otherwise it might be undefined here.

Reply:
You're of course right. The same applies for node_scheduled.

Reply:
Resolved by https://github.com/backdrop/backdrop/pull/2852/commits/c663242b7b579df69144b4c89a3bef8fdb6da662